src/Point/Point.cs

```diff
@@ -81,15 +86,116 @@ public async Task SimulateTrailed()
 
             if (_currentConnection != null)
             {
-                var occupancyStatus = new PointPointPositionMessage(_localId, _remoteId, _pointState.PointPosition, ReportedDegradedPointPosition.PointIsNotInADegradedPosition);
+                var occupancyStatus = new PointPointPositionMessage(_localId, _remoteId, _pointState.PointPosition, _pointState.DegradedPointPosition);
                 await _currentConnection.RequestStream.WriteAsync(new SciPacket() { Message = ByteString.CopyFrom(occupancyStatus.ToByteArray()) });
             }
 
             await UpdateConnectedWebClients();
         }
 
+        private ReportedDegradedPointPosition? GetReportedDegradedPointPosition(ReportedPointPosition previousReportedPosition) => previousReportedPosition switch
+        {
+            ReportedPointPosition.PointIsInARightHandPositionDefinedEndPosition
+                => ReportedDegradedPointPosition.PointIsInADegradedRightHandPosition,
+            ReportedPointPosition.PointIsInALeftHandPositionDefinedEndPosition
+                => ReportedDegradedPointPosition.PointIsInADegradedLeftHandPosition,
+            ReportedPointPosition.PointIsInNoEndPosition
+                => null,
+            ReportedPointPosition.PointIsTrailed
+                => null,
+            _ => null,
+        };
+
+        private ReportedPointPosition? GetReportedPointPositionDegraded(PointPosition pointPosition) => pointPosition switch
+        {
+            PointPosition.NoEndPosition => ReportedPointPosition.PointIsInNoEndPosition,
+            PointPosition.Trailed => ReportedPointPosition.PointIsTrailed,
+            _ => null,
+        };
+
+        private void UpdatePointState(ReportedPointPosition reportedPointPosition, ReportedDegradedPointPosition reportedDegradedPointPosition) {

```

Comment:
so the data types are called Reported... because that's what the telegram calls the variables. Since the values passed to this method are not yet reported to anyone, I'd prefer to omit the 'reported' prefix